### ISSUE TITLE
GCP multi-NIC v1.22.0 acquires license, rolls back

### ISSUE THREAD

**Issue by jfreeland** [F5Networks/f5-declarative-onboarding#248]
### Environment
 * Declarative Onboarding Version: `v1.22.0`
 * BIG-IP Version: `v15`

### Summary
https://github.com/F5Networks/f5-declarative-onboarding/issues/242 was fixed in v1.22.0.

When I run DO on a GCP VM with 3 nics and the mgmt interface is eth1, the node acquires a license from BigIQ, configures itself, and immediately rolls back.

### Steps To Reproduce
1. Create a BigIP node in GCP with 3 network interfaces.  Perform the management interface swap so that your management interface is eth1.

2. Submit a declarative onboarding declaration that acquires a license from BigIQ.

3. Observe the following in restnoded:
```
Tue, 06 Jul 2021 20:44:35 GMT - fine: [f5-declarative-onboarding: restWorker.js] replace localhost /tm/shared/licensing/registration {"licenseText": ...}
Tue, 06 Jul 2021 20:44:35 GMT - finest: [f5-declarative-onboarding: restWorker.js] tryUntil: got error {"code":400}
Tue, 06 Jul 2021 20:44:35 GMT - finest: [f5-declarative-onboarding: restWorker.js] typeof err object
Tue, 06 Jul 2021 20:44:35 GMT - finer: [f5-declarative-onboarding: restWorker.js] tryUntil error: Not a valid F5 License tries left: 5
Tue, 06 Jul 2021 20:44:35 GMT - finest: [f5-declarative-onboarding: restWorker.js] tryUntil: retryOrReject: numRemaining: 5 , code: 400 , message: Not a valid F5 License
Tue, 06 Jul 2021 20:44:35 GMT - finer: [f5-declarative-onboarding: restWorker.js] Unrecoverable error from HTTP request. Not retrying.
Tue, 06 Jul 2021 20:44:35 GMT - info: [f5-declarative-onboarding: restWorker.js] Failed to license: Not a valid F5 License
Tue, 06 Jul 2021 20:44:35 GMT - info: [f5-declarative-onboarding: restWorker.js] Error licensing via BIG-IQ Not a valid F5 License
Tue, 06 Jul 2021 20:44:35 GMT - severe: [f5-declarative-onboarding: systemHandler.js] Error processing system declaration: Not a valid F5 License
Tue, 06 Jul 2021 20:44:35 GMT - severe: [f5-declarative-onboarding: declarationHandler.js] Error processing declaration: Not a valid F5 License
Tue, 06 Jul 2021 20:44:35 GMT - severe: [f5-declarative-onboarding: restWorker.js] Error onboarding: Not a valid F5 License
Tue, 06 Jul 2021 20:44:35 GMT - info: [f5-declarative-onboarding: restWorker.js] Rolling back configuration
Tue, 06 Jul 2021 20:44:35 GMT - fine: [f5-declarative-onboarding: restWorker.js] list localhost /tm/sys/provision
```

### Expected Behavior
To be able to use the license the node acquires.

### Actual Behavior
The node rolls back.

**Comment by dstokesf5** [comment]
So far I have been unable to reproduce this issue. I have a few questions as I continue trying to reproduce this issue:
1. I see the logs saying the license is not valid, have you confirmed that you are using a valid license?
2. Is your BIG-IQ in GCP or somewhere else?

**Comment by jfreeland** [comment]
@dstokesf5 - I'm running into the same issue right now.  Happy to hop on a call and walk you through the example.

**Comment by megamattzilla** [comment]
I'm running into this issue on TMOS 15.1.4 in a 2 NIC deployment in GCP. 

I deployed a 2 NIC Big-IP using the [supported GCP terraform module](https://github.com/F5Networks/terraform-gcp-bigip-module). 

DO 1.19.0 successfully obtains a license from Big-IQ licensing server. 

Redploying the terraform and only changing DO from 1.19 to 1.26 causes DO to fail with invalid license. 

Restnoded log
``` 
/var/log/restnoded/restnoded1.log-Wed, 05 Jan 2022 23:01:01 GMT - finest: [f5-declarative-onboarding: restWorker.js] tryUntil: got error {"code":400}
/var/log/restnoded/restnoded1.log-Wed, 05 Jan 2022 23:01:01 GMT - finest: [f5-declarative-onboarding: restWorker.js] typeof err object
/var/log/restnoded/restnoded1.log-Wed, 05 Jan 2022 23:01:01 GMT - finer: [f5-declarative-onboarding: restWorker.js] tryUntil error: Not a valid F5 License tries left: 5
/var/log/restnoded/restnoded1.log-Wed, 05 Jan 2022 23:01:01 GMT - finest: [f5-declarative-onboarding: restWorker.js] tryUntil: retryOrReject: numRemaining: 5 , code: 400 , message: Not a valid F5 License
/var/log/restnoded/restnoded1.log-Wed, 05 Jan 2022 23:01:01 GMT - finer: [f5-declarative-onboarding: restWorker.js] Unrecoverable error from HTTP request. Not retrying.
/var/log/restnoded/restnoded1.log-Wed, 05 Jan 2022 23:01:01 GMT - info: [f5-declarative-onboarding: restWorker.js] Failed to license: Not a valid F5 License
/var/log/restnoded/restnoded1.log-Wed, 05 Jan 2022 23:01:01 GMT - info: [f5-declarative-onboarding: restWorker.js] Error licensing via BIG-IQ Not a valid F5 License
/var/log/restnoded/restnoded1.log-Wed, 05 Jan 2022 23:01:01 GMT - severe: [f5-declarative-onboarding: systemHandler.js] Error processing system declaration: Not a valid F5 License
/var/log/restnoded/restnoded1.log-Wed, 05 Jan 2022 23:01:01 GMT - severe: [f5-declarative-onboarding: declarationHandler.js] Error processing declaration: Not a valid F5 License
/var/log/restnoded/restnoded1.log-Wed, 05 Jan 2022 23:01:01 GMT - severe: [f5-declarative-onboarding: restWorker.js] Error onboarding: Not a valid F5 License
/var/log/restnoded/restnoded1.log:Wed, 05 Jan 2022 23:01:01 GMT - info: [f5-declarative-onboarding: restWorker.js] Rolling back configuration
```

Big-IQ server shows a license was successfully obtained by the new F5 instance running DO 1.26. 

I'm wondering if this is a regression caused by fixing Issue #242?

**Comment by megamattzilla** [comment]
Adding @meghdivya. Can you please subscribe to this github issue?

**Comment by megamattzilla** [comment]
This is looking to be more of a TMOS issue than specifically DO. When TMOS swaps the MGMT interface to another NIC, the license validation component still checks the mac address of eth0- not eth1 or whichever NIC is now the mgmt interface.   

I replicated this problem without using DO but following the same steps.   

I redeployed a 2 NIC Big-IP into GCP (same terraform as above) and this time licensed Big-IP manually using the [Big-IQ REST API](https://clouddocs.f5.com/products/big-iq/mgmt-api/v6.0/HowToSamples/bigiq_api_modules/t_license_member_management.html#assigning-a-license-to-an-unreachable-device).   

The GCP Big-IP had two network interfaces:  
```
[bigipuser@bigip1:NO LICENSE:Standalone] ~ # tmsh show net interface all-properties

-----------------------------------------------------------------------------------------------------------------------------------
Net::Interface
Name  Status           Mac-Addr   MTU   Bits   Bits  Pkts  Pkts  Mcast  Mcast  Drops  Drops  Errs  Errs  Colli  DoubleTag  DoubleTag     Media      Flow  Trunk    Aggreg
                                          In    Out    In   Out     In    Out     In    Out    In   Out  sions    Pkts In   Pkts Out            Ctrl Req                 
-----------------------------------------------------------------------------------------------------------------------------------
1.0   uninit  42:01:0a:02:00:02  9198      0      0     0     0      0      0      0      0     0     0      0          0          0      none      none   none  detached
mgmt      up  42:01:0a:01:00:02  1500  50.5E  54.8E  6.3E  6.3E   6.3E   6.8E   6.3E   6.2E  6.3E  6.2E  65.9G          0          0  100TX-FD      none   none  detached
```

The NIC swap has alreay occured so interface 1.0 = eth0 and mgmt = eth1.   

I manually assigned an UNREACHABLE license via BIG-IQ using the MAC address of mgmt (eth1) 42:01:0a:01:00:02. The license failed to apply:   

```

[bigipuser@bigip1:ModuleNotLicensed::LICENSE INOPERATIVE:Standalone] config # reloadlic 

Broadcast message from systemd-journald@bigip1.example.com (Mon 2022-01-10 20:38:17 UTC):

mcpd[6596]: 01070608:0: License is not operational (expired or digital signature does not match contents).

2022 Jan 10 20:38:17 bigip1.example.com mcpd[6596]: 01070608:0: License is not operational (expired or digital signature does not match contents)

```

I manually assigned an UNREACHABLE license via BIG-IQ using the MAC address of 1.0 (eth0) 42:01:0a:02:00:02. The license successfully applied:   

```
[bigipuser@bigip1:Active:Standalone] config # tmsh show sys license

Sys::License
Licensed Version                       8.0.0
Registration key                       REDACTED
Licensed On                            2022/01/10
License Start Date                     2022/01/09
License End Date                       2022/02/25
Service Check Date                     2021/08/05
Platform ID                            BIG-IQ-Pool
Daily Renewal Notification Days        5
Daily Renewal Notification Start Date  2022/02/21
```

In DO pre 1.22 it was using the eth0 MAC address for license checkout which was working successfully for Multi-NIC GCP Big-IP devices.   

After 1.22 DO started using the actual MGMT MAC address on multi-NIC GCP Big-IP devices which fails the TMOS license validation.   

TMOS license validation needs to understand NIC swapping in GCP, however, I think #242 should be reverted so that multi-nic license checkout works again in the meantime.

**Comment by dstokesf5** [comment]
Thank you for all of the additional feedback. I have added this issue to our internal product backlog as AUTOTOOL-2612.

**Comment by ghalevy** [comment]
@dstokesf5 - any update on when this will be fixed in a release?

**Comment by dstokesf5** [comment]
This will be resolved in the 1.28.0 release of Declarative Onboarding.

**Comment by dstokesf5** [comment]
This has been resolved in the 1.28.0 release.
